pkg/client/devops/jclient/jenkins.go

```diff
@@ -1,49 +1,28 @@
 package jclient
 
 import (
-	"io/ioutil"
-
-	client "github.com/jenkins-zh/jenkins-client/pkg/core"
-	"gopkg.in/yaml.v2"
+	"github.com/jenkins-zh/jenkins-client/pkg/core"
+	"kubesphere.io/devops/pkg/client/devops/jenkins"
 )
 
 type JenkinsClient struct {
-	
-}
-type Config struct {
-	Jenkins JenkinsConfig `yaml:"devops,omitempty"`
-}
-type JenkinsConfig struct {
-	URL      string `yaml:"host"`
-	UserName string `yaml:"username"`
-	Password string `yaml:"password"`
-	Token    string `yaml:"token,omitempty"`
+	Core core.JenkinsCore
```

Comment:
IMO, `Core core.JenkinsCore` could be replaced by `job.Client` here.

Reply:
So that we can use our Jenkins client conveniently instead of creating `job.Client` every invokation.

Reply:
I agree.

Reply:
That's my mistake, `job.Client` should be  `core.Client`.

Reply:
we can add job.client here instead of replacing the core. What do you think?

Reply:
The `job.Client` is a specific client for a specific function of Jenkins. In jenkins-zh/jenkins-client, there is a [`core.Client`](https://github.com/jenkins-zh/jenkins-client/blob/bc7dc0ac9b3eb3a3ae44ea97dc7f86fffcc33979/pkg/core/core.go#L27-L29) wrapping `core.JenkinsCore`. So we can use it and remove `jclient.JenkinsClient`.

https://github.com/kubesphere/ks-devops/blob/77b5e674169df5edd10bb2d9f167e14de106d0c1/pkg/client/devops/jclient/jenkins.go#L8-L10

Reply:
`jclient.JenkinsClient` also contains functions like `CreateProjectPipeline`, `DeleteProjectPipline`, etc. ([See projectPipeline.go](https://github.com/kubesphere/ks-devops/blob/1cfd8a08c5f16365049395bf9db182a97653715d/pkg/client/devops/jclient/projectPipeline.go)). So, I don't think it can be removed and replaced by core.Client.